docs/testing/contract-tests/contract-tests.md

```diff
@@ -0,0 +1,195 @@
+docker build -t app:build .
+```
+
+**Build Contract Test Docker Images & Execute Tests**
+
+```shell
+docker-compose \
+  -f test-engineering/contract/docker-compose.yml \
+  -p contile-contract-tests \
+  up --abort-on-container-exit --build
+```
+
+### Import Sorting, Linting, Style Guide Enforcement & Static Type Checking
+
+This project uses [Poetry][6] for dependency management. For environment setup, it is recommended to
+use [pyenv][7] and [pyenv-virtualenv][8], as they work nicely with Poetry.
+
+Project dependencies are listed in the `pyproject.toml` file.
+To install the dependencies execute:
+
+```shell
+poetry install
+```
+
+Contributors to this project are expected to execute the following tools.
+Configurations are set in the `pyproject.toml` and `.flake8` files.
+
+**[isort][9]**
+
+ ```shell
+poetry run isort client partner
+ ```
+
+**[black][10]**
+
+ ```shell
+poetry run black client partner
+ ```
+
+**[flake8][11]**
+
+ ```shell
+poetry run flake8 client partner
+ ```
+
+**[mypy][12]**
+
+```shell
+poetry run mypy client partner
+```
+
+## Debugging
+
+The contract testing system is optimized to run within a set of related docker images.
+
+### client
+
+See the `Debugging` section of the client [README][4]
+
+### Contile
+
+To run the contile service, and it's dependent partner service locally, execute the following from
+the contract tests root:
+
+```shell
+docker-compose run -p 8000:8000 contile
+```
+
+Contile runs, by default, on `http://localhost:8000/`. However, it may be necessary to run the tests
+outside of docker, in order to debug functions or manually verify expected results.
+
+**Fetching Tiles**
+
+Contile will attempt to look up your IP address if you access it using Firefox. The easiest way to
+get a test response back would be to craft a curl request. For example (presuming that Contile is
+running on `http://localhost:8000/v1/tiles`):
+
+```sh
+curl -v \
+    -H "User-Agent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:10.0) Gecko/20100101 Firefox/91.0'" \
+    -H "X-Forwarded-For: '89.160.20.115'" \
+    "http://localhost:8000/v1/tiles"
+```
+
+### partner
+
+See the `Local Execution` and `Debugging` sections of the partner [README][5]
+
+## Maintenance
+
+The contract test maintenance schedule cadence is once a quarter and should include
+updating the following:
+
+1. [poetry][6] version and python dependencies
+    * [ ] [pyproject.toml][13]
+    * [ ] [poetry.lock][14]
+2. [Docker][15] artifacts
+    * [ ] client [Dockerfile][16]
+    * [ ] partner [Dockerfile][17]
+    * [ ] [docker-compose.yml][18]
+    * [ ] [docker-compose.204.yml][19]
+    * [ ] [docker-compose.init_error.yml][20]
+    * [ ] [docker-compose.tiles_cache.yml][21]
+3. [CircleCI][22] contract test jobs
+    * [ ] [config.yml][23]
+4. Documentation
+    * [ ] contract tests docs [contract-tests.md][24]
+    * [ ] client docs[client.md][4]
+    * [ ] partner docs [partner.md][5]
+
+[0]: ../../../test-engineering/contract/
```

Comment:
This link is broken, and so are some other ones below.

Reply:
Hrm, both work in pycharm and VSCode. Is it that mdbook does not support absolute paths, as both IDEs resolve the path? I can change those if that's the case.